src/commands/MigrationGenerateCommand.ts

```diff
@@ -258,17 +265,37 @@ ${downSqls.join(`
         timestamp: number,
         upSqls: string[],
         downSqls: string[],
+        esm: boolean,
     ): string {
         const migrationName = `${camelCase(name, true)}${timestamp}`
 
-        return `module.exports = class ${migrationName} {
+        const exportMethod = esm ? "export" : "module.exports ="
+
+        return `/**
+ * @typedef {import('typeorm').QueryRunner} QueryRunner
+ * @typedef {import('typeorm').MigrationInterface} MigrationInterface
+ */
+
+/**
+ * @class
+ * @implements {MigrationInterface}
+ */
+${exportMethod} class ${migrationName} {
     name = '${migrationName}'
 
+    /**
+     * @param {QueryRunner} queryRunner
+     * @returns {Promise<void>}
```

Comment:
If you want to add TSDoc, you can add it in the `MigrationInterface`, not here.

Reply:
I have changed it, `MigrationInterface` should be enough indeed, there is the up and down already defined.
In my Vscode I get better feedback when I add this specific to the up and down, but maybe that is something on my machine or application setup.